src/Lab/Experiments/Silk30InputPrototyping/Api.cs

```diff
@@ -0,0 +1,233 @@
+    MouseButton14,
+    MouseButton15,
+    MouseButton16,
+    MouseButton17,
+    MouseButton18,
+    MouseButton19,
+    MouseButton20,
+    MouseButton21,
+    MouseButton22,
+    MouseButton23,
+    MouseButton24,
+    MouseButton25,
+    MouseButton26,
+    MouseButton27,
+    MouseButton28,
+    MouseButton29,
+    MouseButton30,
+    MouseButton31,
+    A,
+    B,
+    X,
+    Y,
+    LeftBumper,
+    RightBumper,
+    Back,
+    Start,
+    Home,
+    LeftStick,
+    RightStick,
+    DPadUp,
+    DPadRight,
+    DPadDown,
+    DPadLeft
+}
+
+// ---
+// Unchanged
+// ---
+public enum DeadzoneMethod
+{
+    Traditional,
+    AdaptiveGradient
+}
+
+// ---
+// Unchanged
+// ---
+[Flags]
+public enum Position2D
+{
+    Centered = 0,
+    Up = 1,
+    Down = 2,
+    Left = 4,
+    Right = 8,
+    UpLeft = Up | Left,
+    UpRight = Up | Right,
+    DownLeft = Down | Left,
+    DownRight = Down | Right
+}
+
+// ---
+// Unchanged
+// ---
+public enum StandardCursor
+{
+    Default,
+    Arrow,
+    Crosshair,
+    Hand,
+    HResize,
+    VResize
+}
+
+// ---
+// Unchanged
+// ---
+public readonly record struct Axis(int Index, float Position);
+
+// ---
+// RENAMED: from Button
+// ---
+public readonly record struct ButtonState(ButtonName Name, int Index, bool Pressed);
+
+// ---
+// Unchanged
+// ---
+public readonly record struct Deadzone(DeadzoneMethod Method, float Value);
+
+// ---
+// RENAMED: from Hat
+// ---
+public readonly record struct HatState(int Index, Position2D Position);
+
+// ---
+// REMOVED: ScrollWheel (replaced with Vector2)
+// ---
+
+// ---
+// RENAMED: from Thumstick
+// REMOVED: ThumbstickState.X/Y
+// ADDED: ThumbstickState.Position (replaces X/Y)
+// RENAMED: Thumbstick.Position to Thumbstick.Extent
+// ---
+public readonly record struct ThumbstickState(int Index, Vector2 Position)
+{
+    //public float Extent { get; }
+    //public float Direction { get; }
+}
+
+public readonly record struct CursorState(Vector2 Position);
+
+// ---
+// REMOVED: Trigger (replaced with Axis)
+// ---
+
+// ---
+// REFACTOR: IMouse to MouseState - getters moved out to be a immutable record
+// ---
+public readonly partial record struct MouseState
+(
+    IInputDevice Device,
+    ReadOnlyMemory<ButtonState> Buttons,
+    ReadOnlyMemory<Vector2> ScrollWheels,
+    CursorState Cursor
+)
+{
+    public partial bool IsButtonPressed(ButtonName btn);
+}
+
+// ---
+// REFACTOR: ICursor to CursorConfiguration - getters moved out to be a immutable record
+// ---
+public readonly record struct CursorConfiguration
+(
+    StandardCursor? StandardImage = null,
+    ReadOnlyMemory<byte>? CustomImage = null,
+    bool? IsConfined = null,
+    Vector2D<int>? Hotspot = null
+);
+
+// ---
+// REFACTOR: IMouse to MouseConfiguration - settings which aren't state moved out
+// ---
+public readonly record struct MouseConfiguration
+(
+    int? DoubleClickTime = null,
+    int? DoubleClickRange = null,
+    CursorConfiguration? Cursor = null
+);
```

Comment:
Probably worth discussing the benefits of record struct, vs an immutable class with builder pattern in the context of configuration.